packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/editor/services/MdbPluginDisposable.kt

```diff
@@ -33,6 +35,12 @@ class MdbPluginDisposable : Disposable {
      * Disposable interface that gets called when this service is being disposed
      */
     override fun dispose() {
+        // At the very least we would always want to persist our modified settings (ToolbarSettings, PluginSettings)
+        // when closing the project which is why we explicitly invoke StoreUtil.saveSettings to persist them to disk
+        ApplicationManager.getApplication().invokeLater {
+            StoreUtil.saveSettings(ApplicationManager.getApplication())
+        }
```

Comment:
This fixes the problem of ToolbarSettings not getting persisted when a project was closed or application restarted right after making a selection

Reply:
Now that it's a project settings, shouldn't be StoreUtil.saveSettings(project)?

Reply:
it turns out I don't need this anymore - I added this before converting the ToolbarSettings to project level service and now that it is a project level service, it gets disposed off correctly when closing the project.